pkg/stub/stub.go

```diff
@@ -329,6 +327,7 @@ func New(p interface{}, opts ...Option) (Stub, error) {
 
 		registrationTimeout: DefaultRegistrationTimeout,
 		requestTimeout:      DefaultRequestTimeout,
+		logger:              nrilog.Get(),
```

Comment:
@gauravkghildiyal I think it would also make sense to add a `WithLogger(nrilog.Logger) Option` option which would allow setting the used logger with a `stub.Option`.